### ISSUE TITLE
subsetting AFArrays

### ISSUE THREAD

**Issue by alha02** [JuliaComputing/ArrayFire.jl#96]
It would be nice to be able to subset an existing AFArray without having to get the data from the device and then push it back. If I do:
```
keep = trues(AFArray{Bool}, 3, 1)
keep[2] = false
x = randn(AFArray{Float64}, 10, 3)
Array(x[keep, :])
```
I get the following output:
```
3x3 Array{Float64,2}:
 -1.35003  -1.65827  -0.596977
 -1.35003  -1.65827  -0.596977
 -1.35003  -1.65827  -0.596977
```
where
```
julia> Array(x)
10x3 Array{Float64,2}:
 -1.35003   -1.65827    -0.596977
 -0.121518   0.0107386   0.97084
 -0.28531    1.39863     0.155876
 -0.656605  -1.01922    -2.0676
  0.369328   1.10722    -0.0702322
  0.348746   1.10099    -1.01764
 -1.59176    1.01411     1.09976
  1.86908    0.503389    0.692357
 -2.00912   -1.36618    -0.218464
 -0.126166  -0.179565   -1.51902
```
I.e. it is just using the first row values for all subset rows. Is it possible to make the results consistent with subsetting a regular Array?

**Comment by ranjanan** [comment]
Hi @alha02, you can subset AFArrays without transferring back and forth

```julia
julia> keep = trues(AFArray{Bool}, 3, 1)
3-element ArrayFire.AFArray{Bool,1}:
 true
 true
 true

julia> keep[2] = false
false

julia> x = randn(AFArray{Float32}, 10, 3)
10×3 ArrayFire.AFArray{Float32,2}:
 -0.0979873  -0.335571    0.216864
 -0.116751   -0.545407    1.08063 
  0.127151   -0.634063   -1.03452 
  0.915726   -1.98177     0.768012
  0.158653    1.561       0.926395
  0.43059    -0.271434   -2.29116 
 -0.227576    0.274184    0.752289
 -0.870602    1.51401    -0.885461
 -0.144518    1.72336    -1.35225 
  0.0366567  -0.0302384   2.93977 

julia> x[keep, :]
3×3 ArrayFire.AFArray{Float32,2}:
 -0.0979873  -0.335571  0.216864
 -0.0979873  -0.335571  0.216864
 -0.0979873  -0.335571  0.216864
```
Do you face any problems running the above code?

**Comment by alha02** [comment]
Hi ranjanan, thanks for the swift reply. No I don't have an problems running the code, I was just expecting a different result. I just saw that I made a mistake in the example earlier, but if we take this example without ArrayFire:

julia> keep = fill(true, 10)
10-element Array{Bool,1}:
 true
 true
 true
 true
 true
 true
 true
 true
 true
 true

julia> keep[3] = false
false

julia> keep[5] = false
false

julia> x = randn(10, 3)
10√ó3 Array{Float64,2}:
  1.3285      0.591993   0.196063
  1.20153    -0.678173  -0.437311
  1.60228    -0.913024   0.450914
  0.0663512   0.153337   0.657297
  1.00398     1.43451   -0.502761
  0.789337   -0.609603   0.49772
  1.00232    -0.548162  -0.27043
  0.91795     2.27598   -0.507482
  0.969988   -1.91469    0.290196
 -0.328862   -1.77672    0.516662

julia> x[keep, :]
8√ó3 Array{Float64,2}:
  1.3285      0.591993   0.196063
  1.20153    -0.678173  -0.437311
  0.0663512   0.153337   0.657297
  0.789337   -0.609603   0.49772
  1.00232    -0.548162  -0.27043
  0.91795     2.27598   -0.507482
  0.969988   -1.91469    0.290196
 -0.328862   -1.77672    0.516662

I.e. we are getting the rows which were marked as true. Doing the corresponding with AFArrays:

julia> keep = trues(AFArray{Bool}, 10); Array(keep)
10-element Array{Bool,1}:
 true
 true
 true
 true
 true
 true
 true
 true
 true
 true

julia> keep[3] = false
false

julia> keep[5] = false
false

julia> x = randn(AFArray{Float64}, 10, 3); Array(x)
10√ó3 Array{Float64,2}:
  0.495329    0.0439976  -0.0146301
 -1.14184     0.94527     0.0261304
  0.15447     0.0711642  -0.10551
  0.929474   -0.782073   -0.343161
 -0.0909761   0.672831   -0.275725
 -0.203035    0.98724     0.216912
 -0.0869212  -0.852904    0.433115
 -0.0992599  -0.64156    -0.241999
  1.00945     0.891629   -0.589536
  0.642857   -0.532777   -1.03145

julia> Array(x[keep, :])
10√ó3 Array{Float64,2}:
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301
 0.495329  0.0439976  -0.0146301

I.e. we are getting a full-sized matrix where all rows are taken as the first row in the original x
